src/provider.js

```diff
@@ -574,23 +595,32 @@ export default function () {
          *  @description
          *  Update the display options of a SidebarMenuListItem.
          *
-         *  @param {Object} displayOptions Display options you want to update.
-         *  @param {String} displayOptions.title The new title.
-         *  @param {String} displayOptions.prefix The new prefix.
-         *  @param {String} displayOptions.icon The new icon.
-         *  @param {String} displayOptions.iconClass The new iconClass.
-         *  @param {String} displayOptions.category The new category.
-         *  @param {String} displayOptions.status The new status.
-         *  @param {Number|String} itemId The unique id of the item you want to update display.
+         *  @param {Object} displayOptionsParam Display options you want to update.
+         *  @param {String} displayOptionsParam.title The new title.
+         *  @param {String} displayOptionsParam.prefix The new prefix.
+         *  @param {String} displayOptionsParam.icon The new icon.
+         *  @param {String} displayOptionsParam.iconClass The new iconClass.
+         *  @param {String} displayOptionsParam.category The new category.
+         *  @param {String} displayOptionsParam.status The new status.
+         *
+         *  @param {Number|String|Object} criteriaParam The criteria to find the item to update.
+         *  If it's a Number or a String, then it's considered as an id.
          *
          *  @return {SidebarMenuListItem} The founded SidebarMenuListItem with given id.
          */
-    sidebarMenuService.updateItemDisplay = function updateItemDisplay(displayOptionsParam, itemId) {
+    sidebarMenuService.updateItemDisplay = function updateItemDisplay(
+      displayOptionsParam,
+      criteriaParam,
+    ) {
       let displayOptions = displayOptionsParam;
 
-      const item = this.getItemById(itemId);
+      const criteria = _.isString(criteriaParam) || _.isNumber(criteriaParam)
+        ? { id: criteriaParam }
+        : criteriaParam;
+
+      const item = this.getItemByCriteria(criteria);
       if (item) {
-        displayOptions = _.pick(displayOptions, _.identity); // remove falsy attributes
+        displayOptions = _.omit(displayOptions, _.isEmpty); // remove falsy attributes
```

Comment:
If changed this because:
- it didn't work
- it makes more sense to `omit` properties in this context imho :)

Reply:
it's great you took the time to fix it ; i just hope it won't break anything because sidebar code is really messy and sensitive :/ it looks good to me but i'm still not sure that it won't introduce some bad side effects

Would it be possible to have some QC with this pull request, just to be sure it doesn't break the sidebar in one of the universe?

Reply:
Well, this method actually never worked so I don't think it is necessary 😅 

To be honest, the best way to test it would be to have unit testing, but it's not a priority apparently

Reply:
alright if it wasn't working at all i guess it cannot be worst :)

Reply:
I've checked, it's only used 5 times and each time, the method can't find the targeted item, so it wasn't of any use before